src/main/java/com/epam/ta/reportportal/demodata/service/DefaultDemoDataFacade.java

```diff
@@ -96,12 +100,22 @@ public DefaultDemoDataFacade(DemoDataLaunchService demoDataLaunchService, DemoDa
 		User creator = userRepository.findById(user.getUserId())
 				.orElseThrow(() -> new ReportPortalException(ErrorType.USER_NOT_FOUND, user.getUsername()));
 
-		return IntStream.range(0, rq.getLaunchesQuantity()).mapToObj(i -> {
-			Launch launch = demoDataLaunchService.startLaunch(NAME, i, creator, projectDetails);
-			generateSuites(suitesStructure, i, launch.getUuid(), user, projectDetails);
-			demoDataLaunchService.finishLaunch(launch.getUuid());
-			return launch.getId();
-		}).collect(toList());
+		ExecutorService executor = Executors.newFixedThreadPool(rq.getLaunchesQuantity(),
+				new ThreadFactoryBuilder().setNameFormat("demo-data-task-%d").build()
+		);
+
+		List<CompletableFuture<Long>> futures = IntStream.range(0, rq.getLaunchesQuantity())
+				.mapToObj(i -> CompletableFuture.supplyAsync(() -> {
+					Launch launch = demoDataLaunchService.startLaunch(NAME, i, creator, projectDetails);
+					generateSuites(suitesStructure, i, launch.getUuid(), user, projectDetails);
+					demoDataLaunchService.finishLaunch(launch.getUuid());
+					return launch.getId();
+				}, executor))
+				.collect(toList());
+
+		List<Long> generatedLaunchIds = futures.stream().map(CompletableFuture::join).collect(toList());
```

Comment:
You can insert this into upper lambda pipline and get list of long instead of list of completable futures